### ISSUE TITLE
Can not detach the last file from a FileAttachmentField

### ISSUE THREAD

**Issue by maxime-rainville** [unclecheese/silverstripe-dropzone#49]
I think commit d2944d8402ff2a50b89b492bf154efdeb2fa31f3 has introduced a new problem.

If your FileAttachmentField has one or many existing files associated with it and you try to detach all those files, the change doesn't register. You can detach files as long as you leave at least one.

I think the problem is that if you detach all the files in your FileAttachmentField, there's no value at all sent back to the server for the field.

So when you hit the following condition on line 213, it will not update the relation:
```
if(($relation = $this->getRelation()) && is_array($this->Value())) {
    $relation->setByIDList($this->Value());
} elseif($record->has_one($fieldname)) {
...
```

I tried removing `&& is_array($this->Value()` on my local version and I was able to detach all file attachments again.

**Comment by unclecheese** [comment]
Ah, thanks for this. Yeah, there is a bug that appears when you run `setByIDList(null)`. That method needs an array.

Can you change it to:
```
if(($relation = $this->getRelation())) {
    $relation->setByIDList((array) $this->Value());
}
```

... and confirm that works?

**Comment by maxime-rainville** [comment]
@unclecheese

I've added the proposed fix to my local version of the plugin.

I tried doing the following operations:
# Adding files to a DataObject through a frontend FileAttachmentField
# Detach all files through a frontend FileAttachmentField
# Adding files to a DataObject through a backend FileAttachmentField
# Detach all files through a backend FileAttachmentField

I got the expected behaviour in all cases. So the fix appears to work.

**Comment by unclecheese** [comment]
Many thanks. Will make an update.

**Comment by unclecheese** [comment]
Try this? https://github.com/unclecheese/silverstripe-dropzone/commit/9de98bd48a6838c6119247ad5c6ea06c64b56645

**Comment by maxime-rainville** [comment]
I've pull down your change and ran the same test.

Still working fine. Thank you very much.

**Comment by unclecheese** [comment]
Released in `1.2.3`. Thanks again.
